Source/ProgressionSystemRuntime/Public/Data/PSWorldSubsystem.h

```diff
@@ -41,4 +55,44 @@ class PROGRESSIONSYSTEMRUNTIME_API UPSWorldSubsystem : public UWorldSubsystem
 	/** Progression System component reference*/
 	UPROPERTY(VisibleInstanceOnly, BlueprintReadWrite, Transient, Category = "C++", meta = (BlueprintProtected, DisplayName = "Progression System Component"))
 	TObjectPtr<class UPSHUDComponent> ProgressionSystemComponentInternal = nullptr;
+
+	/** Store the current save game instance */
+	UPROPERTY(VisibleInstanceOnly, BlueprintReadWrite, Category = "C++", meta = (BlueprintProtected, DisplayName = "Save Game Instance"))
+	TObjectPtr<class UPSSaveGameData> SaveGameInstanceInternal = nullptr;
+
+	/** The current Saved Progression of a player. */
+	UPROPERTY(VisibleInstanceOnly, BlueprintReadWrite, Category = "C++", meta = (BlueprintProtected, DisplayName = "Saved Progression Row Data"))
+	FPSRowData SavedProgressionRowDataInternal = FPSRowData::EmptyData;
+
+	/** The Progression Data Table that is responsible for progression configuration. */
+	UPROPERTY(VisibleInstanceOnly, BlueprintReadWrite, Transient, Category = "C++", meta = (BlueprintProtected, DisplayName = "Progression Data Table"))
+	TObjectPtr<UDataTable> ProgressionDataTableInternal = nullptr;
```

Comment:
All data tables and assets keep in const pointer to make sure no one is writing there from code in runtime: `TObjectPtr<const UDataTable>`

Reply:
got it, done